rename: hosted_zone_preprocessor.rs to route53_resource_id_preprocess…

Velfi opened the pull request:
## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->
[aws-sdk-rust#554](https://github.com/awslabs/aws-sdk-rust/issues/554)

## Description
<!--- Describe your changes in detail -->
rename: hosted_zone_preprocessor.rs to route53_resource_id_preprocessor.rs
update: trim_hosted_zone to work for more kinds of resource ids
add: new Route53 protocol test for GetChange resource id trimming
update: InlineDependency.forRustFile to accept filenames with a ".rs" extension

## Testing
<!--- Please describe in detail how you tested your changes -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
I added a new protocol test

## Checklist
<!--- If a checkbox below is not applicable, then please DELETE it rather than leaving it unchecked -->
- [x] I have updated `CHANGELOG.next.toml` if I made changes to the smithy-rs codegen or runtime crates
- [x] I have updated `CHANGELOG.next.toml` if I made changes to the AWS SDK, generated SDK code, or SDK runtime crates

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

rcoh commented on aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/customize/route53/Route53Decorator.kt:
```diff
@@ -50,29 +52,39 @@ class Route53Decorator : RustCodegenDecorator {
+            InlineAwsDependency.forRustFile(
+                "route53_resource_id_preprocessor",
+                visibility = Visibility.PRIVATE,
+                CargoDependency.Regex,
```
cargo udeps should hopefully catch this

Velfi replied:
You are correct, I wonder why it didn't catch that.

rcoh commented on aws/sdk/aws-models/route53-tests.smithy:
```diff
@@ -18,3 +18,17 @@ apply ListResourceRecordSets @httpRequestTests([
         }
     }
 ])
+
+apply GetChange @httpRequestTests([
+    {
+        id: "GetChangeTrimChangeId",
+        documentation: "This test validates that change id is correctly trimmed",
+        method: "GET",
+        protocol: "aws.protocols#restXml",
+        uri: "/2013-04-01/change/SOMECHANGEID",
+        bodyMediaType: "application/xml",
+        params: {
+            "Id": "/change/SOMECHANGEID"
+        }
+    }
+])
```
lets add protocol tests for all the supported prefixes

Velfi commented:
I had originally added the `resource` prefix because boto had it but I can't see anywhere in the model where that would be used so I've removed it. I also added a test for `delegationset` so now all 3 supported prefixes are supported
